File: python_mid_june_2024/242. Valid Anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        m, n = len(s), len(t)
        if m != n:
            return False
        sdict = defaultdict(int)
        tdict = defaultdict(int)

        for char in s:
            sdict[char] += 1
        for char in t:
            tdict[char] += 1
        for item in sdict:
            if sdict[item] != tdict[item]:
                return False
        return True

File: java_july/242. Valid Anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()){
            return false;
        }
        int[] freqs = new int[26];

        for (char c : s.toCharArray()){
            freqs[c-'a']++;
        }
        for (char c : t.toCharArray()){
            freqs[c-'a']--;
        }
        for (int freq : freqs) {
            if (freq != 0) {
                return false;
            }
        }
        return true;
    }
}